UnityProject/Assets/Scripts/Items/Tool/AirlockPainter.cs

```diff
@@ -0,0 +1,31 @@
+using Doors;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AirlockPainter : MonoBehaviour
+{
+	[Tooltip("Airlock painting jobs.")]
+	public List<GameObject> AvailablePaintJobs;
+
+	public async void ChoosePainJob(GameObject paintableAirlock)
+	{
+		GameObject chosenPaintJob = await UIManager.RadialMenu.ShowRadialMenu(AvailablePaintJobs, paintableAirlock);
+		if(chosenPaintJob != null)
+		{
+			paintTheAirlock(paintableAirlock, chosenPaintJob);
+		}
+	}
+
+	private void paintTheAirlock(GameObject paintableAirlock, GameObject chosenPaintJob)
```

Comment:
This will need to be called from a command in PlayerNetworkedActions

Reply:
Also do a validation check in the command too, just in case the player has moved away

Reply:
This is a private method of the AirlockPainter class. How can I call it from the PlayerNetworkActions class?

Reply:
Guess you could just move the whole method to PlayerNetworkActions since it doesnt rely on this class or make it public

Reply:
I probably don't understand something again. Wouldn't it look strange that the airlock painting method is in PlayerNetworkActions? Why transfer this method there? Does this solve the interface issue? ))) sorry for the persistence, I just want to figure it out

Reply:
PNA is just our class that holds misc. client -> server requests.

Since it looks like your UI is all client-side, you need a way to inform the server. You could do this through IF2 (interaction system), PNA or a net message. 

PNA is probably fine here.